packages/tlschema/api-report.md

```diff
@@ -707,6 +707,8 @@ export const noteShapeProps: {
     growY: T.Validator<number>;
     url: T.Validator<string>;
     text: T.Validator<string>;
+    userName: T.Validator<string | undefined>;
+    userId: T.Validator<string | undefined>;
```

Comment:
Ideally we'd wait for this (and all other attribution) features until we have user accounts, both as a reason to create an account and because I don't think we can fully deliver the feature without those. But it's worth thinking about how "information about users" can become part of the document, i.e. for offline / downloads.

Perhaps the document can keep track of "known users", or users who have interacted with / created content in the document. That would give us a way to preserve the relationship between userIds and userNames. ie when a user joins or updates their name / color, we also update the document's list of "known users" and preserve that data there. So rather than keeping the name on the shape here, we just use the userId, and if that id is known to the document, then we pull the name from there. I don't know how often userIds are changed (i.e. if I reload my page, do I have a new user id? @ds300) so this may also make the feature not great. We would also want to add this "known users" to copy and paste data, similar to how we do assets, so that it can be brought across between documents.

Reply:
> (i.e. if I reload my page, do I have a new user id? @ds300)

no the user ids are stable per browser context, unless you click 'reset data' at some point.